esi_leap/api/controllers/v1/contract.py

```diff
@@ -102,19 +100,48 @@ def post(self, new_contract):
         return Contract(**c.to_dict())
 
     @wsme_pecan.wsexpose(Contract, wtypes.text)
-    def delete(self, contract_uuid):
+    def delete(self, contract_id):
         request = pecan.request.context
         cdict = request.to_policy_values()
         policy.authorize('esi_leap:contract:delete', cdict, cdict)
 
-        c = contract.Contract.get(request, contract_uuid)
-        if c.project_id != request.project_id:
-            o = offer.Offer.get(request.project_id, c.offer_uuid)
-            if o.project_id != request.project_id:
+        permitted = ContractsController._contract_get_all_authorized_contracts(
+            contract_id, cdict)
+
+        permitted[0].cancel()
+
+    @staticmethod
+    def _contract_get_all_authorized_contracts(contract_id, cdict):
+
+        c_id = contract.Contract.get(contract_id)
+        permitted = []
+        for c in c_id:
+            try:
+                ContractsController._contract_authorize_management(c, cdict)
+                permitted.append(c)
+            except oslo_policy.PolicyNotAuthorized:
+                continue
+
+            if len(permitted) > 1:
+                raise exception.ContractDuplicateName(name=contract_id)
+
+        if len(permitted) == 0:
+            raise exception.ContractNotFound(contract_id=contract_id)
+
+        return permitted
+
+    @staticmethod
+    def _contract_authorize_management(c, cdict):
+
+        if c.project_id != cdict['project_id']:
+            try:
                 policy.authorize('esi_leap:contract:contract_admin',
                                  cdict, cdict)
-
-        c.cancel()
+            except oslo_policy.PolicyNotAuthorized:
+                o = offer.Offer.get_by_uuid(c.offer_uuid)
+                if o.project_id != cdict['project_id']:
+                    policy.authorize('esi_leap:contract:contract_admin',
```

Comment:
Should this check the `offer_admin` policy?

Reply:
Your'e correct. I have changed this.